src/model/entities/segment.ts

```diff
@@ -55,17 +55,11 @@ export class Segment {
 
   public putOnAir(): void {
     this.isSegmentOnAir = true
+    this.executedAtEpochTime ??= Date.now()
```

Comment:
Wouldn't you need to always update the `executedAt` whenever the Segment comes on Air?
I believe this is guarding against if `putOnAir` is called multiple times without being taken off Air. But that sounds more like that `putOnAir` should have an `assertNotOnAir` guard instead of this.

Reply:
Correct. I have updated both putOnAir and takeOffAir to assert that they are off air and on air respectively when called.